modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -785,26 +785,32 @@ public void remove(@NotNull Throwable e) {
                     }
             );
         }).exceptionally(t -> {
-            Throwable cause = t.getCause();
+            Throwable ex;
 
-            if (cause instanceof ConfigurationChangeException) {
-                cause = cause.getCause();
-            }
+            if (t instanceof CompletionException) {
+                if (t.getCause() instanceof ConfigurationChangeException) {
+                    ex = t.getCause().getCause();
+                } else {
+                    ex = t.getCause();
+                }
 
-            if (cause instanceof TableAlreadyExistsException) {
-                TableAlreadyExistsException tableAlreadyExistsException = (TableAlreadyExistsException) cause;
+                assert ex != null;
+            } else {
+                ex = t;
+            }
 
+            if (ex instanceof TableAlreadyExistsException) {
                 tableAsync(name, false).thenAccept(table -> {
                     if (!exceptionWhenExist) {
                         tblFut.complete(table);
                     }
 
-                    removeListener(TableEvent.CREATE, clo, new IgniteInternalCheckedException(tableAlreadyExistsException));
+                    removeListener(TableEvent.CREATE, clo, new IgniteInternalCheckedException(ex));
```

Comment:
Why have you removed the variable?

Reply:
Because we have checked `ex instanceof TableAlreadyExistsException`, variable is not needed.